### ISSUE TITLE
no-jinja-when: not reported if when condition uses multi-line list form

### ISSUE THREAD

**Issue by richm** [ansible/ansible-lint#2959]
##### Summary

no-jinja-when error is not reported if the `when` condition uses the multi-line form

For example - this gives an error:
```yaml
- name: Populate service facts
  service_facts:
  when: "'{{ somevar }} in ansible_facts"
```
but this does not:
```yaml
- name: Populate service facts
  service_facts:
  when:
    - "'{{ somevar }} in ansible_facts"
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.11.0 using ansible 2.14.1
```

##### STEPS TO REPRODUCE

see above

##### Desired Behavior

I expect bad usage of jinja templating to be reported when using valid multi-line `when` conditionals
